assets/quick-add-bulk.js

```diff
@@ -1,212 +1,216 @@
+              this.cleanErrorMessageOnType(event);
+              // Error handling
+              return;
+            }
+
+            this.renderSections(parsedState);
+
+            publish(PUB_SUB_EVENTS.cartUpdate, { source: 'quick-add', cartData: parsedState });
+
+          }).catch((error) => {
+            console.error(error);
+          })
       }
-      const elementToReplace = sectionElement && sectionElement.querySelector(section.selector) ? sectionElement.querySelector(section.selector) : sectionElement;
-      if (elementToReplace) {
-        elementToReplace.innerHTML =
-          this.getSectionInnerHTML(parsedState.sections[section.section], section.selector);
+
+      getSectionsToRender() {
+        return [
+          {
+            id: `quick-add-bulk-${this.dataset.id}-${this.closest('.collection').dataset.id}`,
+            section: this.closest('.collection').dataset.id,
+            selector: `#quick-add-bulk-${this.dataset.id}-${this.closest('.collection').dataset.id}`
+          },
+          {
+            id: 'cart-icon-bubble',
+            section: 'cart-icon-bubble',
+            selector: '.shopify-section'
+          },
+          {
+            id: 'CartDrawer',
+            selector: '#CartDrawer',
+            section: 'cart-drawer'
+          }
+        ];
       }
-    }));
 
-    if (this.isEnterPressed) {
-      this.querySelector(`#Quantity-${this.lastActiveInputId}`).select();
-    }
+      getSectionInnerHTML(html, selector) {
+        return new DOMParser()
+          .parseFromString(html, 'text/html')
+          .querySelector(selector).innerHTML;
+      }
+
+      renderSections(parsedState) {
+        this.getSectionsToRender().forEach((section => {
+          const sectionElement = document.getElementById(section.id);
+          if (sectionElement && sectionElement.parentElement && sectionElement.parentElement.classList.contains('drawer')) {
+            parsedState.items.length > 0 ? sectionElement.parentElement.classList.remove('is-empty') : sectionElement.parentElement.classList.add('is-empty');
+
+            setTimeout(() => {
+              document.querySelector('#CartDrawer-Overlay').addEventListener('click', this.cart.close.bind(this.cart));
+            });
+          }
+          const elementToReplace = sectionElement && sectionElement.querySelector(section.selector) ? sectionElement.querySelector(section.selector) : sectionElement;
+          if (elementToReplace) {
+            elementToReplace.innerHTML =
+              this.getSectionInnerHTML(parsedState.sections[section.section], section.selector);
+          }
+        }));
+
+        if (this.isEnterPressed) {
+          this.querySelector(`#Quantity-${this.lastActiveInputId}`).select();
+        }
 
-    this.listenForActiveInput();
-    this.listenForKeydown();
-  }
+        this.listenForActiveInput();
+        this.listenForKeydown();
+      }
+    }
+);
 }
-customElements.define('quick-add-bulk', QuickAddBulk);
```

Comment:
What's the value in writing the class inline? Like why not wrap this line with the `if (!customElements.get('quick-add-bulk'))` conditional?

Reply:
@dan-menard Fair point! I am just following our existing standard in Dawn where we have done this before: https://github.com/Shopify/dawn/blob/main/assets/localization-form.js#L1

We can always change the approach